### PR TITLE
  [FLOC-2268] Pin ZFS version.

### PR DESCRIPTION
**Pull request by tomprince**
https://clusterhq.atlassian.net/browse/FLOC-2268

Once this is accepted, and the tests have passed, I'll change drop the `--enablerepo clusterhq-testing` and put upload the packages to the release repository.

I've manually tested that the cloud instructions work on CentOS/Rackspace.

Supercedes #1593.

### CODE REVIEW

File: admin/flocker-zfs.spec [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+Conflicts:	zfs-test > %{zfs_version}
+
+%prep
+
+%clean
+
+%files
+
+%description
+This package ensures that a versions of ZFSOnLinux that has been validated
```
Comment (wallrj) [inline admin/flocker-zfs.spec]: typos "a versions" "are installed"

File: admin/flocker-zfs.spec [hunk 2]
```diff
@@ -0,0 +1,46 @@
+# Build this with `rpmbuild -bb admin/flocker-zfs.spec` on the target distribution.
+
+Name:		flocker-zfs
+Version:	0
+Release:	1%{?dist}
+Summary:	Metapackage installing known working version of ZFS.
+
+Group:		System Environment/Kernel
+License:	ASL 2.0
+URL:		http://www.clusterhq.com/
```
Comment (wallrj) [inline admin/flocker-zfs.spec]: :question: Should there be a URL pointing to the repo where the source for this version can be found?

**Comment by wallrj** [comment]
Thanks @tomprince 

 * I spotted a couple of typos
 * Successfully built the metapackage on fedora20 
 * tried installing it and noted that the dependencies all look correct.

```
(release-maintenance/flocker-1.0.0pre1/zfs-meta-FLOC-2268 ? =)[~/.../HybridLogic/flocker]$ sudo yum install  /home/richard/rpmbuild/RPMS/noarch/flocker-zfs-0-1.fc20.noarch.rpm
[sudo] password for richard: 
Loaded plugins: langpacks, refresh-packagekit
Examining /home/richard/rpmbuild/RPMS/noarch/flocker-zfs-0-1.fc20.noarch.rpm: flocker-zfs-0-1.fc20.noarch
Marking /home/richard/rpmbuild/RPMS/noarch/flocker-zfs-0-1.fc20.noarch.rpm to be installed
Resolving Dependencies
--> Running transaction check
---> Package flocker-zfs.noarch 0:0-1.fc20 will be installed
--> Processing Dependency: libnvpair1 = 0.6.3-g7f3e466.clusterhq.0.fc20 for package: flocker-zfs-0-1.fc20.noarch
clusterhq                                                                                                                      | 2.9 kB  00:00:00     
rpmfusion-free-updates                                                                                                         | 3.3 kB  00:00:00     
rpmfusion-nonfree-updates                                                                                                      | 3.3 kB  00:00:00     
updates/20/x86_64/metalink                                                                                                     |  23 kB  00:00:00     
updates                                                                                                                        | 4.9 kB  00:00:00     
virtualbox                                                                                                                     |  951 B  00:00:00     
updates/20/x86_64/primary_db                                                                                                   |  11 MB  00:00:08     
(1/3): rpmfusion-free-updates/20/x86_64/primary_db                                                                             | 434 kB  00:00:05     
(2/3): updates/20/x86_64/updateinfo                                                                                            | 2.0 MB  00:00:06     
(3/3): updates/20/x86_64/pkgtags                                                                                               | 1.5 MB  00:00:00     
--> Processing Dependency: libuutil1 = 0.6.3-g7f3e466.clusterhq.0.fc20 for package: flocker-zfs-0-1.fc20.noarch
--> Processing Dependency: libzfs2 = 0.6.3-g7f3e466.clusterhq.0.fc20 for package: flocker-zfs-0-1.fc20.noarch
--> Processing Dependency: libzpool2 = 0.6.3-g7f3e466.clusterhq.0.fc20 for package: flocker-zfs-0-1.fc20.noarch
--> Processing Dependency: spl = 0.6.3-76_g6ab0866.clusterhq.0.fc20 for package: flocker-zfs-0-1.fc20.noarch
--> Processing Dependency: spl-dkms = 0.6.3-76_g6ab0866.clusterhq.0.fc20 for package: flocker-zfs-0-1.fc20.noarch
--> Processing Dependency: zfs = 0.6.3-g7f3e466.clusterhq.0.fc20 for package: flocker-zfs-0-1.fc20.noarch
--> Processing Dependency: zfs-dkms = 0.6.3-g7f3e466.clusterhq.0.fc20 for package: flocker-zfs-0-1.fc20.noarch
--> Finished Dependency Resolution

```

Please merge

File: admin/flocker-zfs.spec [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+Conflicts:	zfs-test > %{zfs_version}
+
+%prep
+
+%clean
+
+%files
+
+%description
+This package ensures that a version of ZFSOnLinux that has been validated
```
Comment (adamtheturtle) [inline admin/flocker-zfs.spec]: "a package...are installed" > "a package...is installed" (probably not a 1.0 priority)

**Comment by adamtheturtle** [comment]
In two runs of just running the failing Vagrant acceptance tests, I cannot reproduce this locally.

```
(tmp-ec7c083adb060281)~/Documents/ClusterHQ/flocker$ vagrant box add http://build.clusterhq.com/results/vagrant/release-maintenance/flocker-1.0.0pre1/zfs-meta-FLOC-2268/flocker-tutorial.json
(tmp-ec7c083adb060281)~/Documents/ClusterHQ/flocker$ ssh-add ~/.vagrant.d/insecure_private_key
(tmp-ec7c083adb060281)~/Documents/ClusterHQ/flocker$ admin/run-acceptance-tests --distribution centos-7 --provider vagrant --branch=release-maintenance/flocker-1.0.0pre1/zfs-meta-FLOC-2268 --keep --flocker-version=1.0.0pre1-65-g4dfcdb8 flocker.acceptance.test_moving_data.MovingDataTests.test_moving_data flocker.acceptance.test_api.ContainerAPITests.test_move_container_with_dataset
...
flocker.acceptance.test_moving_data
  MovingDataTests
    test_moving_data ... The cluster configuration has been updated. It may take a short while for changes to take effect, in particular if Docker images need to be pulled.
The cluster configuration has been updated. It may take a short while for changes to take effect, in particular if Docker images need to be pulled.
                                                  [OK]
flocker.acceptance.test_api
  ContainerAPITests
    test_move_container_with_dataset ...                                   [OK]

-------------------------------------------------------------------------------
Ran 2 tests in 99.419s

PASSED (successes=2)
```

File: admin/flocker-zfs.spec (hunk 3 above)
Comment (tomprince) [inline admin/flocker-zfs.spec]: No. But also doesn't take any time to fix.

**Comment by itamarst** [comment]
We decided not to do this.

**Pull request closed without merging**